src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java

```diff
@@ -213,4 +325,46 @@ private void checkAuthorizedUpdateUser(User user, DataAccessRequest dar) {
       throw new ForbiddenException("User not authorized to update this Data Access Request");
     }
   }
+
+  private DataAccessRequest updateDarWithDocumentContents(
+      DarDocumentType type,
+      User user,
+      DataAccessRequest dar,
+      InputStream uploadInputStream,
+      FormDataContentDisposition fileDetail) throws IOException {
+    String fileName = fileDetail.getFileName();
+    String toStoreFileName =  UUID.randomUUID() + "." + FilenameUtils.getExtension(fileName);
+    BlobId blobId = gcsService.storeDocument(uploadInputStream, fileDetail.getType(), toStoreFileName);
+    switch (type) {
+      case IRB:
+        // Delete the current document if it exists
+        if (Objects.nonNull(dar.getData().getIrbDocumentLocation())) {
+          deleteDarDocument(dar, dar.getData().getIrbDocumentLocation());
+        }
+        dar.getData().setIrbDocumentLocation(blobId.getName());
+        dar.getData().setIrbDocumentName(fileName);
+        break;
+      case COLLABORATION:
+        // Delete the current document if it exists
+        if (Objects.nonNull(dar.getData().getCollaborationLetterLocation())) {
+          deleteDarDocument(dar, dar.getData().getCollaborationLetterLocation());
+        }
+        dar.getData().setCollaborationLetterLocation(blobId.getName());
+        dar.getData().setCollaborationLetterName(fileName);
+        break;
+      default:
+        break;
+    }
+    dataAccessRequestService.updateByReferenceIdVersion2(user, dar);
```

Comment:
This function already calls findByReferenceId as it's return value so you don't need to call ```dataAccessRequestService.findByReferenceId()``` again

Reply:
Nice catch!